Mathlib/RingTheory/Localization/Algebra.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.RingTheory.Localization.Basic
+import Mathlib.Algebra.Module.LocalizedModule
+import Mathlib.RingTheory.Ideal.Maps
+
+/-!
+# Localization of algebra maps
+
+In this file we provide constructors to localize algebra maps. Also we show that
+localization commutes with taking kernels for ring homomorphisms.
+
+## Implementation detail
+
+The proof that localization commutes with taking kernels does not use the result for linear maps,
+as the translation is currently tedious and can be unified easily after the localization refactor.
+
+-/
+
+universe u u' v v' w w'
+
+variable {R S P : Type*} (Q : Type*) [CommSemiring R] [CommSemiring S] [CommSemiring P]
+  [CommSemiring Q]
+  {M : Submonoid R} {T : Submonoid P}
+  [Algebra R S] [Algebra P Q] [IsLocalization M S] [IsLocalization T Q]
+  (g : R →+* P)
+
+variable (S) in
+/-- The canonical linear map from the kernel of `g` to the kernel of its localization. -/
+def RingHom.toKerIsLocalization (hy : M ≤ Submonoid.comap g T) :
+    RingHom.ker g →ₗ[R] RingHom.ker (IsLocalization.map Q g hy : S →+* Q) where
+  toFun x := ⟨algebraMap R S x, by simp [RingHom.mem_ker, (RingHom.mem_ker g).mp x.property]⟩
+  map_add' x y := by
+    simp only [AddSubmonoid.coe_add, Submodule.coe_toAddSubmonoid, map_add, AddSubmonoid.mk_add_mk]
+  map_smul' a x := by
+    simp only [SetLike.val_smul, smul_eq_mul, map_mul, RingHom.id_apply,
+      SetLike.mk_smul_of_tower_mk, Algebra.smul_def]
+
+@[simp]
+lemma RingHom.toKerIsLocalization_apply (hy : M ≤ Submonoid.comap g T) (r : RingHom.ker g) :
+    (RingHom.toKerIsLocalization S Q g hy r).val = algebraMap R S r :=
+  rfl
+
```

Comment:
IMO a better way here is (needs `import Mathlib.RingTheory.Localization.Ideal`)
```
lemma IsLocalization.mk'_mem_map_algebraMap (I : Ideal R) (x : R) (s : M) :
    IsLocalization.mk' S x s ∈ I.map (algebraMap R S) ↔ ∃ s ∈ M, s * x ∈ I := by
  rw [← Ideal.unit_mul_mem_iff_mem _ (IsLocalization.map_units S s), IsLocalization.mk'_spec',
    IsLocalization.mem_map_algebraMap_iff M]
  simp_rw [← map_mul, IsLocalization.eq_iff_exists M, mul_comm x, ← mul_assoc, ← Submonoid.coe_mul]
  exact ⟨fun ⟨⟨y, t⟩, c, h⟩ ↦ ⟨_, (c * t).2, h ▸ I.mul_mem_left c.1 y.2⟩, fun ⟨s, hs, h⟩ ↦
    ⟨⟨⟨_, h⟩, ⟨s, hs⟩⟩, 1, by simp⟩⟩

lemma IsLocalization.ker_map (hT : Submonoid.map g M = T) :
    RingHom.ker (IsLocalization.map Q g (hT.symm ▸ M.le_comap_map) : S →+* Q) =
      (RingHom.ker g).map (algebraMap R S) := by
  ext x
  obtain ⟨x, s, rfl⟩ := IsLocalization.mk'_surjective M x
  simp [RingHom.mem_ker, IsLocalization.map_mk', IsLocalization.mk'_eq_zero_iff,
    IsLocalization.mk'_mem_map_algebraMap, ← hT]
```
and then show that `I.map (algebraMap R S)` is the localization of `I` (unsure if we have it or not).

Reply:
Please check if what I did was what you wanted.